roles/edpm_deploy_baremetal/tasks/main.yml

```diff
@@ -88,14 +88,119 @@
+          'dataplane',
+          'cr'
+        ] | ansible.builtin.path_join
+      }}
+  environment:
+    PATH: "{{ cifmw_path }}"
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+  block:
+    - name: Perform kustomizations to the OpenStackDataPlaneNodeSet CR
+      when: not cifmw_edpm_deploy_baremetal_dry_run
+      cifmw.general.ci_kustomize:
+        target_path: "{{ cifmw_edpm_deploy_openstack_crs_path }}"
+        sort_ascending: false
+        kustomizations: |-
+          apiVersion: kustomize.config.k8s.io/v1beta1
+          kind: Kustomization
+          patches:
+            - target:
+                kind: OpenStackDataPlaneNodeSet
+              patch: |-
+          {% if content_provider_registry_ip is defined %}
+                - op: add
+                  path: /spec/nodeTemplate/ansible/ansibleVars/edpm_container_registry_insecure_registries
+                  value: ["{{ content_provider_registry_ip }}:5001"]
+          {% endif %}
+
+                - op: add
+                  path: /spec/nodeTemplate/ansible/ansibleVars/edpm_bootstrap_command
+                  value: sudo dnf -y update
+        kustomizations_paths: >-
+          {{
+            [
+              (
+                [
+                  cifmw_edpm_deploy_baremetal_manifests_dir,
+                  'kustomizations',
+                  'dataplane'
+                ] | ansible.builtin.path_join
+              )
+            ]
+          }}
+      register: cifmw_edpm_deploy_baremetal_crs_kustomize_result
+
+- name: Log the CR that is about to be applied
+  when: not cifmw_edpm_deploy_baremetal_dry_run
+  ansible.builtin.debug:
+    var: cifmw_edpm_deploy_baremetal_crs_kustomize_result
+
+- name: Patch OpenStackDataPlaneNodeSet to add repo-setup-downstream service
+  when:
+    - cifmw_edpm_deploy_baremetal_repo_setup_override
+    - cifmw_edpm_deploy_baremetal_create_vms | bool
+    - not cifmw_edpm_deploy_baremetal_dry_run
+  block:
+    # This file will be created in downstream job's pre-playbook
+    - name: Create repo-setup-downstream OpenStackDataPlaneService
+      environment:
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+        PATH: "{{ cifmw_path }}"
+      ansible.builtin.command:
+        cmd: >-
+          oc apply -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
+          -f "{{ cifmw_installyamls_repos }}/devsetup/edpm/services/dataplane_v1beta1_openstackdataplaneservice_reposetup_downstream.yaml"
+    # to-do: We can drop this step once we drop dev-preview#1 jobs in downstream
+    # This is added because install_yamls is tagged and we don't
+    # have repo-setup service in OpenStackDataPlane in v0.1.0 tag
+    - name: Get list of services defined under OpenStackDataPlaneNodeSet resource
+      environment:
+        PATH: "{{ cifmw_path }}"
+      ansible.builtin.command:
+        cmd: >-
+          yq '.spec.services[]' {{ cifmw_edpm_deploy_baremetal_crs_kustomize_result.output_path }}
```

Comment:
could not we use `cifmw.general.ci_kustomize` for this? Also, I think these downstream tasks could possibly be dropped as the downsteam repos are expected to be added in `bootstrap` role using `bootstrap_command`.

Reply:
I was trying to follow the previous pattern of getting the list of services and add/replace depending on the list